dex-it/src/test/scala/com/wavesplatform/it/sync/DEXClientFaultToleranceTestSuite.scala

```diff
@@ -3,63 +3,75 @@ package com.wavesplatform.it.sync
+class DEXClientFaultToleranceTestSuite extends NewMatcherSuiteBase {
+
+  val toxiproxy: ToxiProxy  = new ToxiProxy().start()
+  val proxy: ContainerProxy = toxiproxy.container.getProxy(WavesNodeContainerInfo.netAlias, WavesNodeContainerInfo.dexGrpcExtensionPort)
+
+  override protected def dexInitialSuiteConfig: Config = {
+    ConfigFactory.parseString(s"""waves.dex {
+         |  price-assets = [ "$UsdId", "WAVES" ]
+         |  grpc.integration {
+         |    waves-node-grpc {
+         |      host = ${toxiproxy.name}
+         |      port = ${toxiproxy.getInnerProxyPort(proxy)}
+         |    }
+         |  }
+         |}""".stripMargin)
   }
 
-  private val wavesNode2Container: Coeval[WavesNodeContainer] = Coeval.evalOnce { createWavesNode("waves-2") }
+  private val wavesNode2Container: Coeval[base.WavesNodeContainer] = Coeval.evalOnce { createWavesNode("waves-2") }
 
-  protected val cachedWavesNode2ApiAddress = CachedData {
-    dockerClient.getExternalSocketAddress(wavesNode2Container(), wavesNode2Container().restApiPort)
-  }
+  protected val cachedWavesNode2ApiAddress = CachedData(wavesNode2Container().restApiAddress)
 
   private def wavesNode2Api: NodeApi[Id] = fp.sync(NodeApi[Try]("integration-test-rest-api", cachedWavesNode2ApiAddress.get()))
 
   override protected def beforeAll(): Unit = {
-    startAndWait(wavesNode1Container(), wavesNode1Api)
+    startAndWait(wavesNodeContainer(), wavesNodeApi)
     broadcastAndAwait(IssueUsdTx)
-    startAndWait(dex1Container(), dex1Api)
+    startAndWait(dexContainer(), dexApi)
   }
 
   "DEXClient should works correctly despite of the short connection losses" in {
+
     val aliceBuyOrder = mkOrder(alice, wavesUsdPair, OrderType.BUY, 1.waves, 300)
 
-    lazy val alice2BobTransferTx = mkTransfer(alice, bob, amount = wavesNode1Api.balance(alice, usd), asset = usd)
-    lazy val bob2AliceTransferTx = mkTransfer(bob, alice, amount = wavesNode1Api.balance(bob, usd), asset = usd)
+    lazy val alice2BobTransferTx = mkTransfer(alice, bob, amount = wavesNodeApi.balance(alice, usd), asset = usd)
+    lazy val bob2AliceTransferTx = mkTransfer(bob, alice, amount = wavesNodeApi.balance(bob, usd), asset = usd)
 
     markup("Alice places order that requires some amount of USD, DEX receives balances stream from the node 1")
-    dex1Api.place(aliceBuyOrder)
-    dex1Api.waitForOrderStatus(aliceBuyOrder, OrderStatus.Accepted)
+    dexApi.place(aliceBuyOrder)
+    dexApi.waitForOrderStatus(aliceBuyOrder, OrderStatus.Accepted)
 
     markup(s"Disconnect DEX from the network and perform USD transfer from Alice to Bob")
-    dockerClient.disconnectFromNetwork(dex1Container)
+    proxy.setConnectionCut(true)
 
-    broadcastAndAwait(wavesNode1Api, alice2BobTransferTx)
-    usdBalancesShouldBe(wavesNode1Api, 0, defaultAssetQuantity)
+    broadcastAndAwait(wavesNodeApi, alice2BobTransferTx)
+    usdBalancesShouldBe(wavesNodeApi, 0, defaultAssetQuantity)
 
-    Thread.sleep(2000)
+    Thread.sleep(5000)
```

Comment:
could we exclude this method of waiting from tests?

Reply:
We use this in order to simulate 5 seconds connection loss

Reply:
Done